src/cpp/parse.cpp

```diff
@@ -5615,7 +5615,7 @@ bool Parser::rTypeNameOrFunctionType(typet &tname)
       if(!rArgDeclaration(parameter_declaration))
         return false;
 
-      code_typet::parametert parameter;
+      code_typet::parametert parameter(typet{});
```

Comment:
`(std::move(parameter_declaration))`?

Reply:
That would have been nice, but these are (to the compiler) unrelated types, both inheriting from `exprt`.